File: 0090-subsets-ii/0090-subsets-ii.py
class Solution:
    def subsetsWithDup(self, nums: List[int]) -> List[List[int]]:
        res = []
        nums.sort()

        def loop(a, i):
            res.append(a[:])
            for j in range(i, len(nums)):
                if j > i and nums[j] == nums[j - 1]:
                    continue
                a.append(nums[j])
                loop(a, j + 1)
                a.pop()
        loop([], 0)
        return res

File: 0090-subsets-ii/0090-subsets-ii.js
/**
 * @param {number[]} nums
 * @return {number[][]}
 */
var subsetsWithDup = function(nums) {
    const res = [];
    nums.sort((a, b) => a - b);
    function loop(a, i) {
        res.push([...a]);
        for (let j = i; j < nums.length; j++) {
            if (j > i && nums[j] === nums[j - 1]) continue;
            a.push(nums[j]);
            loop(a, j + 1);
            a.pop()

        }
    }

    loop([], 0);
    return res;
};